src/material-experimental/mdc-button/button-base.ts

```diff
@@ -39,6 +39,8 @@ export const MAT_BUTTON_HOST = {
   // Add a class that applies to all buttons. This makes it easier to target if somebody
   // wants to target all Material buttons.
   '[class.mat-mdc-button-base]': 'true',
+  '(focus)': '_focused = true',
+  '(blur)': '_focused = false',
```

Comment:
Note that this strategy will run into this bug: https://github.com/angular/components/issues/19402